src/DimensionReduction/pca.jl

```diff
@@ -25,5 +25,6 @@ function fit(ds::DataSet, pca::PCA)
     catch
         error("No local descriptors found in DataSet")
     end
-    λ, W = select_eigendirections(d .- mean.(d), pca.tol)
+    # dc = [di - mean(d) for di in d]
+    λ, W = select_eigendirections(d, pca.tol)
```

Comment:
I think it is necessary to subtract the mean. [Here](
https://github.com/cesmix-mit/PotentialLearning.jl/blob/4bac004a64da1772550c3edba7a6de6d1d90f283/examples/ACE/utils/PotentialLearningExtensions.jl#L93) is how I did it in the variant of this code I implemented.

Reply:
The function `transform` should also be impacted by the mean. See [here](https://github.com/cesmix-mit/PotentialLearning.jl/blob/4bac004a64da1772550c3edba7a6de6d1d90f283/examples/ACE/utils/PotentialLearningExtensions.jl#L102).

Reply:
Yes, this is how I understand PCA to work as well. However, in your `transform!()` function, technically we should be centering w.r.t the training mean (i.e. `pca.m`), not computing a new mean for the descriptors. See this [stackoverflow discusssion](https://stats.stackexchange.com/questions/142216/zero-centering-the-testing-set-after-pca-on-the-training-set) and how they do it with [scikit](https://github.com/scikit-learn/scikit-learn/blob/364c77e047ca08a95862becf40a04fe9d4cd2c98/sklearn/decomposition/_base.py#L100), i.e., they use the mean stored in the data class obtained on from the fitting data.

Reply:
I have some other concerns though about how to use PCA correctly, especially when your fitting data come from configurations with different numbers of atoms (i.e., the, energy distribution would be multimodal). I think we should merge this PR as is, and move this PCA stuff to a separate branch/issue.